.tekton/main-pipeline.yaml

```diff
@@ -175,79 +160,162 @@ spec:
+          OCI repository. An empty string means the artifacts do not expire.
+        type: string
+        default: ""
+      results:
+      - name: SOURCE_ARTIFACT
+        description: The Trusted Artifact URI pointing to the artifact with
+          the smuggled activation key.
+        type: string
+      volumes:
+        - name: workdir
+          emptyDir: {}
+      stepTemplate:
+        volumeMounts:
+          - mountPath: /var/workdir
+            name: workdir
       steps:
+      - name: use-trusted-artifact
+        image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:4e39fb97f4444c2946944482df47b39c5bbc195c54c6560b0647635f553ab23d
+        args:
+          - use
+          - $(params.SOURCE_ARTIFACT)=/var/workdir/source
       # TODO(ROX-20651): use content sets instead of subscription manager for access to RHEL RPMs once available.
       - name: smuggle-activation-key
         image: registry.access.redhat.com/ubi8/ubi:latest
-        script: exec "$(workspaces.source.path)/source/.konflux/scripts/subscription-manager/subscription-manager-bro.sh" smuggle
+        workingDir: /var/workdir/source
+        script: .konflux/scripts/subscription-manager/subscription-manager-bro.sh smuggle
+      - name: create-trusted-artifact
+        image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:4e39fb97f4444c2946944482df47b39c5bbc195c54c6560b0647635f553ab23d
+        args:
+          - create
+          - --store
+          - $(params.ociStorage)
+          - $(results.SOURCE_ARTIFACT.path)=/var/workdir/source
+        env:
+          - name: IMAGE_EXPIRES_AFTER
+            value: $(params.ociArtifactExpiresAfter)
 
   - name: fetch-external-networks
-    runAfter:
-    - determine-image-tag
+    params:
+      - name: SOURCE_ARTIFACT
+        value: $(tasks.copy-subscription-manager-activation-key.results.SOURCE_ARTIFACT)
+      - name: ociStorage
+        value: $(params.output-image-repo):konflux-$(params.revision).external-networks
+      - name: ociArtifactExpiresAfter
+        value: $(params.oci-artifact-expires-after)
     taskSpec:
+      params:
+      - name: SOURCE_ARTIFACT
+        description: The Trusted Artifact URI pointing to the artifact with
+          the application source code. This should be the result of the subscription-manager task,
+          to ensure order of execution.
+        type: string
+      - name: ociStorage
+        description: The OCI repository where the Trusted Artifacts are stored.
+        type: string
+      - name: ociArtifactExpiresAfter
+        description: Expiration date for the trusted artifacts created in the
+          OCI repository. An empty string means the artifacts do not expire.
+        type: string
+        default: ""
+      results:
+      - name: SOURCE_ARTIFACT
+        description: The Trusted Artifact URI pointing to the artifact with
+          the external networks.
```

Comment:
```suggestion
        description: The Trusted Artifact URI pointing to the application source code with
          the additional external networks definitions.
```

Reply:
Done in 15566365d11efcbb1d8563313489922ed725117a